samza-azure/src/main/java/org/apache/samza/system/eventhub/admin/EventHubSystemAdmin.java

```diff
@@ -77,49 +77,31 @@ private String getNextOffset(String currentOffset) {
-    final Map<String, CompletableFuture<EventHubRuntimeInformation>> ehRuntimeInfos = new HashMap<>();
+    Map<String, SystemStreamMetadata> requestedMetadata = new HashMap<>();
 
-    streamNames.forEach((streamName) -> {
+    try {
+      for (String streamName : streamNames) {
         if (!streamPartitions.containsKey(streamName)) {
+          LOG.debug(String.format("Partition ids for Stream=%s not found", streamName));
+
           EventHubClientManager eventHubClientManager = getOrCreateStreamEventHubClient(streamName);
-          CompletableFuture<EventHubRuntimeInformation> runtimeInfo = eventHubClientManager.getEventHubClient()
-                  .getRuntimeInformation();
+          CompletableFuture<EventHubRuntimeInformation> runtimeInfo = eventHubClientManager.getEventHubClient().getRuntimeInformation();
 
-          ehRuntimeInfos.put(streamName, runtimeInfo);
+          long timeoutMs = eventHubConfig.getRuntimeInfoWaitTimeMS(systemName);
+          EventHubRuntimeInformation ehInfo = runtimeInfo.get(timeoutMs, TimeUnit.MILLISECONDS);
+          LOG.debug(String.format("Adding partition ids for Stream=%s", streamName));
+          streamPartitions.put(streamName, ehInfo.getPartitionIds());
         }
-      });
-
-    try {
-      ehRuntimeInfos.forEach((streamName, ehRuntimeInfo) -> {
-
-          if (!streamPartitions.containsKey(streamName)) {
-            LOG.debug(String.format("Partition ids for Stream=%s not found", streamName));
-            try {
-
-              long timeoutMs = eventHubConfig.getRuntimeInfoWaitTimeMS(systemName);
-              EventHubRuntimeInformation ehInfo = ehRuntimeInfo.get(timeoutMs, TimeUnit.MILLISECONDS);
-
-              LOG.debug(String.format("Adding partition ids for Stream=%s", streamName));
-              streamPartitions.put(streamName, ehInfo.getPartitionIds());
-            } catch (InterruptedException | ExecutionException | TimeoutException e) {
-
-              String msg = String.format("Error while fetching EventHubRuntimeInfo for System:%s, Stream:%s", systemName,
-                  streamName);
-              LOG.error(msg, e);
-              throw new SamzaException(msg, e);
-            }
-          }
-
-          String[] partitionIds = streamPartitions.get(streamName);
-          Map<Partition, SystemStreamPartitionMetadata> sspMetadataMap = getPartitionMetadata(streamName, partitionIds);
-          SystemStreamMetadata systemStreamMetadata = new SystemStreamMetadata(streamName, sspMetadataMap);
-
-          requestedMetadata.put(streamName, systemStreamMetadata);
-        });
-
+        String[] partitionIds = streamPartitions.get(streamName);
+        Map<Partition, SystemStreamPartitionMetadata> sspMetadataMap = getPartitionMetadata(streamName, partitionIds);
+        SystemStreamMetadata systemStreamMetadata = new SystemStreamMetadata(streamName, sspMetadataMap);
+        requestedMetadata.put(streamName, systemStreamMetadata);
```

Comment:
Added the fetched partitionIds as a part of log line.
I think it's better to report the latency through EventHubSystemMetrics rather than logging it here. The metrics reporting is required for all API's of this class. Will create a followup JIRA.

Reply:
+1 to adding a metric for long-running operations.